File: TakaBankSystem.js
let accounts = [];

function showOutput(message, isError = false) {
    const outputBox = document.getElementById('outputBox');
    
    // মেসেজের ধরন অনুযায়ী আইকন পরিবর্তন
    let icon = isError ? '<i class="fa-solid fa-triangle-exclamation" style="color: #ef4444;"></i>' : '<i class="fa-solid fa-circle-check" style="color: #10b981;"></i>';
    
    outputBox.innerHTML = `${icon} <div>${message}</div>`;
    
    // হাইলাইট করার জন্য এনিমেশন ইফেক্ট
    outputBox.style.transform = "scale(1.02)";
    setTimeout(() => {
        outputBox.style.transform = "scale(1)";
    }, 200);
}

function createAccount() {
    let accNum = parseInt(document.getElementById('createAccNum').value);
    let name = document.getElementById('createName').value;
    let initialBalance = parseFloat(document.getElementById('createBalance').value);

    if (isNaN(accNum) || name.trim() === "" || isNaN(initialBalance) || initialBalance < 0) {
        showOutput("Please enter valid details to create an account.", true);
        return;
    }

    let exists = accounts.find(acc => acc.accountNumber === accNum);
    if (exists) {
        showOutput("Account number already exists! Try a different one.", true);
        return;
    }

    let newAccount = {
        accountNumber: accNum,
        accountHolderName: name,
        balance: initialBalance
    };

    accounts.push(newAccount);
    showOutput(`<strong>Account Created Successfully!</strong><br>Name: ${name} <br>A/C No: ${accNum} <br>Balance: ${initialBalance.toFixed(2)} BDT`);
    
    document.getElementById('createAccNum').value = '';
    document.getElementById('createName').value = '';
    document.getElementById('createBalance').value = '';
}

function transaction(type) {
    let accNum = parseInt(document.getElementById('transAccNum').value);
    let amount = parseFloat(document.getElementById('transAmount').value);

    if (isNaN(accNum) || isNaN(amount) || amount <= 0) {
        showOutput("Invalid Account Number or Amount!", true);
        return;
    }

    let account = accounts.find(acc => acc.accountNumber === accNum);

    if (account) {
        if (type === 'deposit') {
            account.balance += amount;
            showOutput(`<strong>Deposit Successful!</strong><br>A/C No: ${accNum} <br>Deposited: ${amount.toFixed(2)} BDT <br>New Balance: ${account.balance.toFixed(2)} BDT`);
        } else if (type === 'withdraw') {
            if (amount <= account.balance) {
                account.balance -= amount;
                showOutput(`<strong>Withdrawal Successful!</strong><br>A/C No: ${accNum} <br>Withdrawn: ${amount.toFixed(2)} BDT <br>Remaining Balance: ${account.balance.toFixed(2)} BDT`);
            } else {
                showOutput(`<strong>Insufficient Balance!</strong><br>Cannot withdraw ${amount.toFixed(2)} BDT.`, true);
            }
        }
        document.getElementById('transAmount').value = ''; // শুধু অ্যামাউন্ট ক্লিয়ার হবে
    } else {
        showOutput("Account not found! Please check the Account Number.", true);
    }
}

function displayAccount() {
    let accNum = parseInt(document.getElementById('dispAccNum').value);

    if (isNaN(accNum)) {
        showOutput("Please enter a valid Account Number.", true);
        return;
    }

    let account = accounts.find(acc => acc.accountNumber === accNum);

    if (account) {
        showOutput(`<strong>--- Account Details ---</strong><br>Account Number: ${account.accountNumber}<br>Account Holder: ${account.accountHolderName}<br>Current Balance: <strong>${account.balance.toFixed(2)} BDT</strong>`);
    } else {
        showOutput("Account not found! Please check the Account Number.", true);
    }
}

File: TakaBankSystem.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>
using namespace std;
class BankAccount {
private:
    int accountNumber;
    string accountHolderName;
    double balance;

public:
    BankAccount(int accNum, string name, double initialBalance) {
        accountNumber = accNum;
        accountHolderName = name;
        balance = initialBalance;
    }

    int getAccountNumber() {
        return accountNumber;
    }

    double getBalance() {
        return balance;
    }

    void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
            cout << "Deposit successful! Current Balance: " << fixed << setprecision(2) << balance << " BDT\n";
        } else {
            cout << "Invalid deposit amount!\n";
        }
    }

    void withdraw(double amount) {
        if (amount > 0 && amount <= balance) {
            balance -= amount;
            cout << "Withdrawal successful! Remaining Balance: " << fixed << setprecision(2) << balance << " BDT\n";
        } else if (amount > balance) {
            cout << "Insufficient Balance! Cannot withdraw " << amount << " BDT.\n";
        } else {
            cout << "Invalid withdrawal amount!\n";
        }
    }

    void displayAccount() {
        cout << "\n--- Account Details ---\n";
        cout << "Account Number: " << accountNumber << "\n";
        cout << "Account Holder: " << accountHolderName << "\n";
        cout << "Balance: " << fixed << setprecision(2) << balance << " BDT\n";
        cout << "-----------------------\n";
    }
};

int main() {
    vector<BankAccount> accounts;
    int choice;

    cout << "=== Welcome to the Bank Management System ===\n";

    while (true) {
        cout << "\n1. Create Account\n";
        cout << "2. Deposit Money\n";
        cout << "3. Withdraw Money\n";
        cout << "4. Display Account\n";
        cout << "5. Exit\n";
        cout << "Enter your choice: ";
        cin >> choice;

        if (choice == 1) {
            int accNum;
            string name;
            double initialDeposit;
            
            cout << "Enter Account Number: ";
            cin >> accNum;
            cout << "Enter Account Holder Name: ";
            cin.ignore();
            getline(cin, name);
            cout << "Enter Initial Deposit (in BDT): ";
            cin >> initialDeposit;

            BankAccount newAccount(accNum, name, initialDeposit);
            accounts.push_back(newAccount);
            cout << "Account created successfully!\n";
        } 
        else if (choice == 2) {
            int accNum;
            double amount;
            cout << "Enter Account Number: ";
            cin >> accNum;
            bool found = false;

            for (auto &acc : accounts) {
                if (acc.getAccountNumber() == accNum) {
                    cout << "Enter Deposit Amount (in BDT): ";
                    cin >> amount;
                    acc.deposit(amount);
                    found = true;
                    break;
                }
            }

            if (!found) {
                cout << "Account not found!\n";
            }
        } 
        else if (choice == 3) {
            int accNum;
            double amount;
            cout << "Enter Account Number: ";
            cin >> accNum;
            bool found = false;

            for (auto &acc : accounts) {
                if (acc.getAccountNumber() == accNum) {
                    cout << "Enter Withdrawal Amount (in BDT): ";
                    cin >> amount;
                    acc.withdraw(amount);
                    found = true;
                    break;
                }
            }

            if (!found) {
                cout << "Account not found!\n";
            }
        } 
        else if (choice == 4) {
            int accNum;
            cout << "Enter Account Number: ";
            cin >> accNum;
            bool found = false;

            for (auto &acc : accounts) {
                if (acc.getAccountNumber() == accNum) {
                    acc.displayAccount();
                    found = true;
                    break;
                }
            }

            if (!found) {
                cout << "Account not found!\n";
            }
        } 
        else if (choice == 5) {
            cout << "Exiting the system. Thank you!\n";
            break;
        } 
        else {
            cout << "Invalid choice! Please try again.\n";
        }
    }

    return 0;
}

